### ISSUE TITLE
'Alt' modifier key not being detected when used alone on linux

### ISSUE THREAD

**Issue by alisterburt** [napari/napari#5612]
## 🐛 Bug

The 'Alt' modifier key is not being detected in a mouse callback on a Linux system here, printing `event.modifiers` shows an empty tuple. This works as expected on my mac. 

Interestingly, 'Control' + 'Alt' is working absolutely fine on the same linux box

## To Reproduce

```python
import napari

def callback(viewer, event):
    print(event.modifiers)

viewer = napari.Viewer()
viewer.mouse_drag_callbacks.append(callback)

napari.run()
```

## Environment

lg7-1g_vchandra> napari --info
napari: 0.4.17
Platform: Linux-4.18.0-425.13.1.el8_7.x86_64-x86_64-with-glibc2.28
Python: 3.10.7 (main, Sep  9 2022, 12:08:13) [GCC 12.1.0]
Qt: 5.15.2
PyQt5: 5.15.9
NumPy: 1.24.2
SciPy: 1.10.1
Dask: 2023.2.0
VisPy: 0.11.0
magicgui: 0.6.1
superqt: unknown
in-n-out: 0.1.6
app-model: 0.1.1
npe2: 0.6.2

OpenGL:
 - GL version:  4.6.0 NVIDIA 470.86
 - MAX_TEXTURE_SIZE: 16384

Screens:
 - screen 1: resolution 1920x1080, scale 1.0

Plugins:
 - napari: 0.4.17 (77 contributions)
 - napari-console: 0.0.7 (0 contributions)
 - napari-svg: 0.1.6 (2 contributions)
 - napari-threedee: 0.0.3 (18 contributions)

## Additional context

reported by Vish Chandrasekaran @ichbintux

**Comment by alisterburt** [comment]
@brisvag @kevinyamauchi could you see if you can reproduce on your system?

**Comment by Czaki** [comment]
On my system, the Left Alt is working, but the Right one is not detected.

**Comment by alisterburt** [comment]
Thanks @Czaki - this is good to know

**Comment by brisvag** [comment]
*lol*, On my machine both left and right `Alt` work just fine xD

**Comment by Czaki** [comment]
Maybe it is somehow connected with the keyboard layout? 
In polish the right alt is used to enter special characters (ąęćźżł) 

In my case, even left alt is marked as alt on the keyboard layout (`ALt L`), when the right alt is recognized as `Level 3 S...` 

![obraz](https://user-images.githubusercontent.com/3826210/224047080-320e1b30-9906-4822-bf7b-20de1bcbe52e.png)

Could @ichbintux share the keyboard layout?

**Comment by ichbintux** [comment]
[image: keybrd.png][image: layout.png]

**Comment by alisterburt** [comment]
@ichbintux images didn't come through, you can copy them directly into comments on GitHub

**Comment by ichbintux** [comment]
![PXL_20230309_162049725 MP](https://user-images.githubusercontent.com/127400430/224104052-f3f8d5b0-103b-4a5d-aa1c-541476006bfa.jpg)
![PXL_20230309_162139566 MP](https://user-images.githubusercontent.com/127400430/224104060-e7f4f712-42b3-46e3-a8c1-4a5747449b4f.jpg)
Thanks for looking into this

**Comment by Czaki** [comment]
And in your case the problem happens when you press left alt or when right alt? As I see that you have different descriptions at the keyboard layout.

**Comment by alisterburt** [comment]
@Czaki it was the left alt

**Comment by Czaki** [comment]
Ok. Maybe toomorow I will prepare code sample with registering all events to handle what happens. 
Generalnie @ichbintux will need to play a little and pastę log.

**Comment by alisterburt** [comment]
Thank you @Czaki - you're incredible

**Comment by ichbintux** [comment]
I thought neither Alt was registering, @alisterburt

**Comment by alisterburt** [comment]
True, neither registered alone - only the left registered when pressed with 'Control'

**Comment by Czaki** [comment]
Please try with this:

```python
from typing import cast

from qtpy.QtWidgets import QApplication
from qtpy.QtCore import QObject, QEvent, Qt
from qtpy.QtGui import QKeyEvent

import napari

def callback(viewer, event):
    print("cal", event.modifiers)

def print_modif(value):
    res = []
    if value & Qt.KeyboardModifier.AltModifier:
        res.append("Alt")
    if value & Qt.KeyboardModifier.ControlModifier:
        res.append("Control")
    if value & Qt.KeyboardModifier.MetaModifier:
        res.append("Meta")
    if value & Qt.KeyboardModifier.ShiftModifier:
        res.append("Shift")
    if value & Qt.KeyboardModifier.GroupSwitchModifier:
        res.append("GroupSwitch")
    if value & Qt.KeyboardModifier.KeypadModifier:
        res.append("Keypad")
    return ", ".join(res)

class KeyEvents(QObject):
    def eventFilter(self, obj, event):
        if event.type() == QEvent.KeyPress:
            event = cast(QKeyEvent, event)
            print(f"Key pressed {event.key()}, {event.text()}; {print_modif(event.modifiers())}")
            return True

        return super().eventFilter(obj, event)


viewer = napari.Viewer()
viewer.mouse_drag_callbacks.append(callback)

app = QApplication.instance()

fil = KeyEvents()
app.installEventFilter(fil)
viewer.window.qt_viewer.installEventFilter(fil)

napari.run()
```

also please check if the standalone Alt click does not activate the menu bar. Which Window Manager is used?

**Comment by brisvag** [comment]
For me, all combinations work. No distinction between left and right Alt, Ctrl and Shift. Here's outputs of pressing first left then right keys, then some combinations

```
Key pressed 16777249, ; Control
Key pressed 16777249, ; Control
Key pressed 16777251, ; Alt
Key pressed 16777251, ; Alt
Key pressed 16777248, ; Shift
Key pressed 16777248, ; Shift
Key pressed 16777248, ; Control, Shift
Key pressed 16777251, ; Alt, Control
Key pressed 16777251, ; Alt, Control

```

**Comment by alisterburt** [comment]
Thanks for the snippet @Czaki - I will be able to test from the workstation in question on Monday

**Comment by ichbintux** [comment]
Sorry for the late reply. All combinations seem to be working for me as
well, now with the new code.
Thanks!
--------
Key pressed 16777251, ; Alt
Key pressed 16777251, ; Alt
Key pressed 16777249, ; Control
Key pressed 16777249, ; Control
Key pressed 16777249, ; Control
Key pressed 16777251, ; Alt
Key pressed 16777251, ; Alt
Key pressed 16777249, ; Control
Key pressed 16777248, ; Shift
Key pressed 16777248, ; Shift
Key pressed 16781571, ; GroupSwitch
Key pressed 16781571, ; GroupSwitch
Key pressed 16781571, ; GroupSwitch
Key pressed 16777249, ; Control
Key pressed 16777249, ; Control
Key pressed 16777249, ; Control
Key pressed 16777248, ; Shift
Key pressed 16777248, ; Shift
Key pressed 16777249, ; Control
Key pressed 16777251, ; Alt, Control
Key pressed 16777248, ; Shift
Key pressed 16777250, ; Meta, Shift
Key pressed 16777248, ; Shift
Key pressed 16777250, ; Meta, Shift
Key pressed 16777299, ;

**Comment by Czaki** [comment]
So my code properly report pushed Keys but they are not properly propagate into callbacks?

**Comment by alisterburt** [comment]
it seems like it @Czaki - @ichbintux was this on the same machine/in the same environment as we ran into issues?

**Comment by ichbintux** [comment]
Yes, it was.

**Comment by psobolewskiPhD** [comment]
@ichbintux Can y'all clarify the status of this issue?
Was it just a EU keyboard layout issue and not a napari issue?
Can we close?

**Comment by alisterburt** [comment]
I think the issue is some linux desktop environments override alt + click to perform window dragging
